subsys/net/lib/http/http_server.c

```diff
@@ -0,0 +1,1508 @@
+/*
+ * Copyright (c) 2023, Emna Rekik
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "headers/http_service.h"
+#include "headers/server_functions.h"
+
+#include <errno.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <strings.h>
+
+#include <zephyr/logging/log.h>
+#include <zephyr/net/http/service.h>
+#include <zephyr/net/tls_credentials.h>
+LOG_MODULE_REGISTER(net_http_server, CONFIG_NET_HTTP_SERVER_LOG_LEVEL);
+
+#define HTTP_SERVER_MAX_RESPONSE_SIZE CONFIG_NET_HTTP_SERVER_MAX_RESPONSE_SIZE
+#define HTTP_SERVER_MAX_URL_LENGTH    CONFIG_NET_HTTP_SERVER_MAX_URL_LENGTH
+#define HTTP_SERVER_MAX_FRAME_SIZE    CONFIG_NET_HTTP_SERVER_MAX_FRAME_SIZE
+
+#define TEMP_BUF_LEN 64
+
+#include <zephyr/kernel.h>
+#include <zephyr/net/net_ip.h>
+#include <zephyr/net/socket.h>
+#include <zephyr/posix/sys/eventfd.h>
+
+#include "../../ip/net_private.h"
+
+static const char preface[] = "PRI * HTTP/2.0\r\n\r\nSM\r\n\r\n";
+
+static const char final_chunk[] = "0\r\n\r\n";
+static const char *crlf = &final_chunk[3];
+
+static const unsigned char settings_frame[9] = {
+	0x00, 0x00, 0x00,        /* Length */
+	0x04,                    /* Type: 0x04 - setting frames for config or acknowledgment */
+	0x00,                    /* Flags: 0x00 - unused flags */
+	0x00, 0x00, 0x00, 0x00}; /* Reserved, Stream Identifier: 0x00 - overall connection */
+
+static const unsigned char settings_ack[9] = {
+	0x00, 0x00, 0x00,        /* Length */
+	0x04,                    /* Type: 0x04 - setting frames for config or acknowledgment */
+	0x01,                    /* Flags: 0x01 - ACK */
+	0x00, 0x00, 0x00, 0x00}; /* Reserved, Stream Identifier */
+
+static const char content_404[] = {
+#ifdef INCLUDE_HTML_CONTENT
+#include "not_found_page.html.gz.inc"
+#endif
+};
+
+#define INVALID_SOCK -1
+
+int http_server_init(struct http_server_ctx *ctx)
+{
+	int proto;
+	int failed = 0, count = 0;
+	int svc_count;
+	socklen_t len;
+	int fd, af, i;
+	struct sockaddr_storage addr_storage;
+	const union {
+		struct sockaddr *addr;
+		struct sockaddr_in *addr4;
+		struct sockaddr_in6 *addr6;
+	} addr = {
+		.addr = (struct sockaddr *)&addr_storage
+	};
+
+	HTTP_SERVICE_COUNT(&svc_count);
+
+	/* Initialize fds */
+	memset(ctx->fds, 0, sizeof(ctx->fds));
+	memset(ctx->clients, 0, sizeof(ctx->clients));
+
+	for (i = 0; i < ARRAY_SIZE(ctx->fds); i++) {
+		ctx->fds[i].fd = INVALID_SOCK;
+	}
+
+	/* Create an eventfd that can be used to trigger events during polling */
+	fd = eventfd(0, 0);
+	if (fd < 0) {
+		fd = -errno;
+		LOG_ERR("eventfd failed (%d)", fd);
+		return fd;
+	}
+
+	ctx->fds[count].fd = fd;
+	ctx->fds[count].events = POLLIN;
+	count++;
+
+	HTTP_SERVICE_FOREACH(svc) {
+		/* set the default address (in6addr_any / INADDR_ANY are all 0) */
+		memset(&addr_storage, sizeof(struct sockaddr_storage), 0);
```

Comment:
Should be `memset(&addr_storage, 0, sizeof(struct sockaddr_storage));`

I guess armgcc didn't throw error here, so it went unnoticed. Also the GitHub CI finds it now too.

Reply:
Indeed, I don't understand how I missed that :facepalm: